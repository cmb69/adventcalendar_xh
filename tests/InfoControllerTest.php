<?php

/*
 * Copyright 2023 Christoph M. Becker
 *
 * This file is part of Adventcalendar_XH.
 *
 * Adventcalendar_XH is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * Adventcalendar_XH is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Adventcalendar_XH.  If not, see <http://www.gnu.org/licenses/>.
 */

namespace Adventcalendar;

use Adventcalendar\Infra\Repository;
use Adventcalendar\Infra\SystemChecker;
use Adventcalendar\Infra\View;
use ApprovalTests\Approvals;
use PHPUnit\Framework\TestCase;

class InfoControllerTest extends TestCase
{
    public function setUp(): void
    {
        global $plugin_tx;

        $plugin_tx = XH_includeVar("./languages/en.php", "plugin_tx");
    }

    public function testRendersPluginInfo()
    {
        $repository = $this->createMock(Repository::class);
        $systemChecker = $this->createMock(SystemChecker::class);
        $systemChecker->method("checkVersion")->willReturn(false);
        $systemChecker->method("checkExtension")->willReturn(false);
        $systemChecker->method("checkPlugin")->willReturn(false);
        $systemChecker->method("checkWritability")->willReturn(false);
        $view = new View("./views/", XH_includeVar("./languages/en.php", "plugin_tx")["adventcalendar"]);
        $sut = new InfoController("./plugins/adventcalendar/", $repository, $systemChecker, $view);
        $response = $sut->defaultAction();
        Approvals::verifyHtml($response);
    }
}
