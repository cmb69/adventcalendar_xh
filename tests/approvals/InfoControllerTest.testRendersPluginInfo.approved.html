<h1>Adventcalendar 1.0beta6</h1>
<div>
    <h2>System check</h2>
    <p class="xh_fail">PHP version is at least 5.4.0: failure</p>
    <p class="xh_fail">PHP extension 'gd' is loaded: failure</p>
    <p class="xh_fail">CMSimple_XH version is at least 1.6.3: failure</p>
    <p class="xh_fail">CMSimple_XH plugin 'jquery' is installed: failure</p>
    <p class="xh_warning">'./plugins/adventcalendar/config/' is writable: warning</p>
    <p class="xh_warning">'./plugins/adventcalendar/css/' is writable: warning</p>
    <p class="xh_warning">'./plugins/adventcalendar/languages/' is writable: warning</p>
    <p class="xh_warning">'' is writable: warning</p>
</div>
